### PR TITLE
SIL pass names. Allow command line options to specific the pass tag.

### PR DESCRIPTION
**Pull request by atrick**
A pass has an ID (C++ identifier), Tag (shell identifier),
and Name (human identifier).

Command line options that identify passes should obviously be compatibile with
with the pass' command line identifier. This is also what the user is used to
typing for the -debug-only option.

### CODE REVIEW

**Comment by atrick** [comment]
@swift-ci smoke test.

**Comment by gottesmm** [comment]
Can you add tests that this behavior is preserved?

**Comment by atrick** [comment]
@swift-ci smoke test and merge.

**Comment by atrick** [comment]
@swift-ci smoke test and merge.

**Comment by gottesmm** [comment]
LGTM

**Review by gottesmm — APPROVED** [review]
LGTM!

File: test/SILOptimizer/pass_printer.swift [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// RUN: %target-swift-frontend -Xllvm -sil-print-before=definite-init -emit-sil %s -o /dev/null 2>&1 | %FileCheck --check-prefix=BEFORE %s
+// RUN: %target-swift-frontend -Xllvm -sil-print-after=definite-init -emit-sil %s -o /dev/null 2>&1 | %FileCheck --check-prefix=AFTER %s
+// RUN: %target-swift-frontend -Xllvm -sil-disable-pass=definite-init -Xllvm -sil-print-pass-name -emit-sil %s -o /dev/null 2>&1 | %FileCheck --check-prefix=DISABLE %s
+
+// BEFORE: SIL function before Guaranteed Passes Definite Initialization for Diagnostics (definite-init) (#1)
```
Comment (gottesmm) [inline test/SILOptimizer/pass_printer.swift]: Actually question. I forgot, does this work when assertions are disabled? You may want a requires line in such a case.

**Comment by atrick** [comment]
@gottesmm actually, none of it is currently under NDEBUG, even the dbgs() output.

**Comment by atrick** [comment]
@swift-ci smoke test and merge.

**Pull request merged**